src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -2989,6 +2991,56 @@ int SketchObject::trim(int GeoId, const Base::Vector3d& point)
         delConstraints(delete_list, false);
     };
 
+    // Pick the appropriate portion which should inherit the point-on-object constraint
+    // (or delete it altogether if it's not on any curve)
+    // Assume that GeoId1 is the pre-existing curve
+    // GeoId2 can be same as GeoId1 (for example for periodic B-spline)
+    auto chooseCurveForPointOnObject = [this](int GeoId1,

```

Comment:
Any reason why you are using a lambda rather than a normal function here? This Lambda is huge and so the function containing it even more so.

Reply:
I could define a normal function, but it would still have to be within this `trim` method, since it's use is limited and arguments are very dependent on `trim`'s internals. Admittedly not the best approach.

Reply:
Oh it's in trim(), I have to say I didn't look in which function it was.
Well then it sounds even less acceptable. Trim function is a huge beast of 700 lines. I'm not a huge advocate of refactoring, but even I can see that 700 is too long.
Perhaps trim should become a class then if you feel that the lambda cannot be taken out.

Reply:
Maybe refactoring trim is another PR though. I'm not against leaving this like that for this PR.

Reply:
Yes. I'll go with this option. This lambda and a lot of code from `trim` can be reused in, say, `split`. All that is best done with some tests added beforehand.